engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/preparation/JobPreparationActor.scala

```diff
@@ -331,12 +333,31 @@ class JobPreparationActor(workflowDescriptor: EngineWorkflowDescriptor,
     import RuntimeAttributeDefinition.{addDefaultsToAttributes, evaluateRuntimeAttributes}
     val curriedAddDefaultsToAttributes =
       addDefaultsToAttributes(runtimeAttributeDefinitions, workflowDescriptor.backendDescriptor.workflowOptions) _
+
+    def applyDockerMirroring(attributes: Map[LocallyQualifiedName, WomValue]): Map[LocallyQualifiedName, WomValue] = {
+      val mirroredImageAttribute = for {
+        mirror <- dockerMirroring
+        origDockerStr <- attributes.get(RuntimeAttributesKeys.DockerKey).map(_.valueString)
+        origDockerImg <- DockerImageIdentifier.fromString(origDockerStr) match {
+          case Success(i) => Some(i)
+          case Failure(e) =>
+            workflowLogger.warn(s"Failed to attempt mirroring image ${origDockerStr} due to ${e.toString}")
+            None
```

Comment:
I wonder if it's better to actively fail or fall back to not mirroring. I doubt it matters because failing to massage strings should be rare.

Reply:
What do you think actively failing would look like here? It doesn't feel like the whole task should be failed, though it probably will be by another part of the system if the provided Docker string is nonsense.

Reply:
It would probably be due to a configuration error, so maybe emit an... `.error(...)` and carry on?

This is the first time in recent memory I've advocated for a _higher_ log level 😆

Reply:
I think we only get this error if the user passes us an invalid Docker image name, it's the `DockerImageIdentifier` string parsing that is failing.